lib/utils/call.jl

```diff
@@ -165,6 +165,12 @@ render_arg(io, arg::Union{<:Base.RefValue, AbstractArray}) = summary(io, arg)
 
 # TODO: replace with JuliaLang/julia#49933 once merged
 
+# note that this is generally only safe with functions that do not call back into Julia.
+# when callbacks occur, the code should ensure the GC is not running:
+# - on 1.10 and later, everything is fine because of safepoint_on_entry
+# - on 1.9, @cfunction-based callbacks are fine because they transition to gc_unsafe
+# - on 1.8 and earlier, the code should explicitly call GC.safepoint()!
```

Comment:
On 1.9+ callbacks from C automatically transition the runtime to GC_unsafe
On 1.8 the code should explicitly transition to GC_unsafe